ngraph/core/src/preprocess/pre_post_process.cpp

```diff
@@ -194,26 +271,90 @@ std::shared_ptr<Function> PrePostProcessor::build(const std::shared_ptr<Function
-            new_param->get_rt_info()["LAYOUT"] =
-                std::make_shared<VariantWrapper<Layout>>(input->m_tensor_data->m_layout);
+        if (!input->m_tensor_data->is_element_type_set()) {
+            // Reuse element type from network
+            input->m_tensor_data->set_element_type(param->get_element_type());
+        }
+        std::vector<std::shared_ptr<ov::Node>> nodes;
+        std::vector<std::shared_ptr<op::v0::Parameter>> new_params;
+        if (input->m_tensor_data->m_color_format == ColorFormat::NV12_TWO_PLANES) {
+            // TODO: use spatial dimensions when set
+            auto shape_y = param->get_partial_shape();
+            auto shape_uv = param->get_partial_shape();
+            // TODO: check network layout
+            shape_y[1] = 1;  // TODO: hardcoded as NCHW for now
+            shape_uv[1] = 1;
+            shape_uv[2] = shape_uv[2].get_length() / 2;
+            auto param_y = std::make_shared<op::v0::Parameter>(input->m_tensor_data->get_element_type(), shape_y);
+            if (!input->m_tensor_data->m_planes_sub_names.empty()) {
+                param_y->set_friendly_name(param->get_friendly_name() + "/" +
+                                           input->m_tensor_data->m_planes_sub_names[0]);
+            } else {
+                param_y->set_friendly_name(param->get_friendly_name() + "/Y");
+            }
+            new_params.push_back(param_y);
+            nodes.push_back(param_y);
+            auto param_uv = std::make_shared<op::v0::Parameter>(input->m_tensor_data->get_element_type(), shape_uv);
+            if (input->m_tensor_data->m_planes_sub_names.size() == 2) {
+                param_uv->set_friendly_name(param->get_friendly_name() + "/" +
+                                            input->m_tensor_data->m_planes_sub_names[1]);
+            } else {
+                param_uv->set_friendly_name(param->get_friendly_name() + "/UV");
+            }
+            new_params.push_back(param_uv);
+            nodes.push_back(param_uv);
+
+        } else if (input->m_tensor_data->m_color_format == ColorFormat::NV12_SINGLE_PLANE) {
+            auto new_param_shape = param->get_partial_shape();
+            // OPENVINO_ASSERT(new_param_shape.is_dy)
+            // TODO: use spatial dimensions
+            if (new_param_shape.rank().is_static() && new_param_shape.rank().get_length() == 4) {
+                // TODO: check network layout
+                new_param_shape = {new_param_shape[0], 1, new_param_shape[2].get_length() * 3 / 2, new_param_shape[3]};
+            }
+            auto new_param =
+                std::make_shared<op::v0::Parameter>(input->m_tensor_data->get_element_type(), new_param_shape);
+            // Old param will be removed, so friendly name can be reused
+            new_param->set_friendly_name(param->get_friendly_name());
+            new_params.push_back(new_param);
+            nodes.push_back(new_param);
```

Comment:
looks like we duplicate the logic from validate and infer type of the operation. Can we take this information after we create operation and just take output information from the operation?